pkg/datarepair/queue/queue.go

```diff
@@ -60,14 +62,20 @@ func (q *Queue) Enqueue(qi *pb.InjuredSegment) error {
 	if err != nil {
 		return queueError.New("error adding injured seg to queue %s", err)
 	}
+	q.cond.Signal()
 	return nil
 }
 
 // Dequeue returns the next repair segement and removes it from the queue
 func (q *Queue) Dequeue() (pb.InjuredSegment, error) {
```

Comment:
For these I usually include a non-blocking version `TryDeque`.

Reply:
line 52. Instead of inputting the system time, can we input the network time, which is most accurate, that way we dont have to worry about what the user has set his system time to be. Not sure if BIOS time can be used here. Generally BIOS time is not changeable after system as booted up.  Just a thought.

Reply:
line 63. If we are unable to put the injured segment into the queue, 1. should we handle it to find out why we are unable to put into the queue and try a recovery mechanism to find out why and try again to put into the queue? 2. if is the first segment that we are trying to put into the queue and it failed, should we issue a cond.Signal()?

Reply:
The implementation shouldn't use time at all.

Reply:
Generally what is the time line between a segment put in the queue for repair till it is actually handled to be repaired? Is it seconds, minutes, days? Reason I am thinking time could be useful is, if it is been long in the queue, that means the piece-nodes that are listed as offline could have come online and  vice-versa. At that time, if we go with old data, then we can potentially replicating the data and still not repair the real lost segments.

Is my thinking right.. lets say , when the segment is ready to be repaired, should the repairer do a ping test and find out at that instance what pieces-nodes are offline and generate only those pieces? basically doing another quick check....what do you think? the reason repairer doing a ping test is because with the time data in side the packet, he can determine if he needs to do a ping test or not. is this a scenario we should consider during repair...

Reply:
It's hard to tell, but I generally think they should be handled fast. Well the repairer tries to download as many pieces it can, hence it acts like a ping anyways. Similarly the checker shouldn't consider a piece lost immediately the node goes down, but rather some threshold (e.g. time it takes to do few restarts). Since the nodes are going down sparingly, this means to get to the repair point, there has to have been quite a few nodes offline for a long duration of time... So at repair time, there's definitely things to repair.